application/models/Project_model.php

```diff
@@ -561,29 +565,26 @@ function get_projects_by_reader_status($params)
     function get_projects_by_author_status($params)
     {
         $status_where = ($params['status'] == 'complete') ? ' AND p.status = "complete" ': ' AND p.status != "complete" ';
-        $sql = '
-                SELECT p.id, p.project_type, p.title_prefix, p.title, p.url_librivox, p.status, p.coverart_thumbnail, p.zip_url, p.zip_size, l.two_letter_code 
-                FROM projects p 
-                JOIN project_authors pa ON (pa.project_id = p.id) 
-                JOIN languages l ON (l.id = p.language_id)
-                WHERE pa.author_id = ' . $params['author_id'] . 
-                $status_where;                 
-
-        $sql .= '
-                UNION
-                SELECT p.id, p.project_type, p.title_prefix, CONCAT(s.title, " (in ", p.title, ")") as title, p.url_librivox, p.status, p.coverart_thumbnail, p.zip_url, p.zip_size, l.two_letter_code
-                FROM sections s 
-                JOIN authors a ON (a.id = s.author_id)
-                JOIN projects p ON (p.id = s.project_id)
-                JOIN languages l ON (l.id = p.language_id)
-                WHERE s.author_id = ' . $params['author_id'] . ' ' .
-                $status_where . '
-                AND p.is_compilation = 1 '; 
+        $sql = sprintf(
+            'SELECT p.id, p.project_type, p.title_prefix, p.title, p.url_librivox, p.status, p.coverart_thumbnail, p.zip_url, p.zip_size, l.two_letter_code
+            FROM projects p
+            JOIN project_authors pa ON (pa.project_id = p.id)
+            JOIN languages l ON (l.id = p.language_id)
+            WHERE pa.author_id = %d %s',
+            $params['author_id'], $status_where);
+
+        $sql .= sprintf(
+            'UNION
+            SELECT p.id, p.project_type, p.title_prefix, CONCAT(s.title, " (in ", p.title, ")") as title, p.url_librivox, p.status, p.coverart_thumbnail, p.zip_url, p.zip_size, l.two_letter_code
+            FROM sections s
+            JOIN authors a ON (a.id = s.author_id)
+            JOIN projects p ON (p.id = s.project_id)
+            JOIN languages l ON (l.id = p.language_id)
+            WHERE s.author_id = %d %s AND p.is_compilation = 1
+            ORDER BY 4 ASC',
+            $params['authod_id'], $status_where);
```

Comment:
```suggestion
            $params['author_id'], $status_where);
```

Reply:
Right, I understand the concept. I'm assuming sprintf will throw an error if you try to feed a non-decimal to %d. Wherever I use %s the replacing string isn't user-provided.

But yeah, I'm not sure why I didn't use ? in this case for author_id, will fix.

Reply:
ah yep, the `%d` is a nice quick fix, and I _think_ it closes most loopholes. It won't throw an error though, because php. 😬  https://onlinephp.io/c/6a733